torch/distributed/launch.py

```diff
@@ -0,0 +1,172 @@
+well-improved single-node training performance. It can also be used in
+multi-node distributed training, by spawning up multiple processes on each node
+for well-improved multi-node distributed training performance as well.
+This will especially be benefitial for systems with multiple Infiniband
+interfaces since all of them can be utilized for aggregated communication
+bandwidth.
+
+In both cases, this utilily will launch a given number of processes
+per node (nproc_per_node, which defaults to the number of GPUs on the node).
+This number needs to be less or euqal to the number of GPUs on the current
+system, and each process will be operating on a single GPU from GPU 0 to
+GPU nproc_per_node - 1.
+
+How to use this module:
+
+    (1) Single Node multi-proc distributed training:
+        python -m torch.distributed.launch YOUR_TRAINING_SCRIPT.py (--arg1
+        --arg2 --arg3 and all other arguments of your training script)
+
+    (2) Multi Node multi-proc distributed training: (e.g. two nodes)
+
+    NODE1: (IP: 192.168.1.1, and has a free port: 1234)
+        python -m torch.distributed.launch --nproc_per_node=NUM_GPUS_YOU_HAVE
+        --num_node=2 --rank_node=0 --master_addr="192.168.1.1"
+        --master_port=1234 YOUR_TRAINING_SCRIPT.py (--arg1 --arg2 --arg3 and
+        all other arguments of your training script)
+    NODE2:
+        python -m torch.distributed.launch --nproc_per_node=NUM_GPUS_YOU_HAVE
+        --num_node=2 --rank_node=1 --master_addr="192.168.1.1"
+        --master_port=1234 YOUR_TRAINING_SCRIPT.py (--arg1 --arg2 --arg3 and
+        all other arguments of your training script)
+
+    (3) To look up what optional arguments this module offers:
+        python -m torch.distributed.launch --help
+
+Important Notices:
+
+(1) This utilty and multi-process distributed (single node or
+multi-node) GPU training currently only achieves the best performance using
+the NCCL distributed backend. Thus NCCL backend is the recommended backend to
+use for GPU training.
+
+(2) In your training program, you are supposed to parse the command-line
+argument: --device=DEVICE_TO_RUN, (which will be provided with this module) and
+ensure that your code only runs on this device in your training program by:
+
+    torch.cuda.set_device(arg.device)  # before your code runs
+
+    or
+
+    with torch.cuda.device(arg.device):
+        # your code to run
+
+(3) In your training program, you are supposed to call the following function
+at the beginning to start the distributed backend. You need to make sure that
+the init_method uses "env://", which is the only supported init_method by this
+module:
+
+    torch.distributed.init_process_group(backend='YOUR BACKEND',
+                                         "init_method='env://')
+
+(4) In your training program, you are supposed to convert your model to
+DistributedDataParallel module using the following function. Please ensure
```

Comment:
Well it's not bound to `DistributedDataParallel` in any way. We can say that you can use regular distributed functions, but if you want to use `DistributedDataParallel`, then here's how to configure it.